tweaks.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# For more information, please refer to <http://unlicense.org>
+
+"""
+This module lists game-specific tweaks, that cannot be handled by other means
+(specifically tweaks, that are outside of dxvk.conf or Mesa scope, such as
+additional environment variables, renaming files post-installation, changing
+Wine settings for the specific game, etc.).  The purpose is similar to
+"Install Scripts" *.vfd files distributed with games, except here tweaks are
+provided by Proton community and not game developer.
+"""
+
+
+TWEAKS_DB = {
+    # Call of Duty® (2003)
+    "2620": {
+        "env": {
+            "MESA_EXTENSION_MAX_YEAR": "2003",
+            "__GL_ExtensionStringVersion": "17700",
+        }
+    },
+    # STAR WARS™ Jedi Knight - Jedi Academy™
+    "6020": {
+        "env": {
+            "MESA_EXTENSION_MAX_YEAR": "2003",
+            "__GL_ExtensionStringVersion": "17700",
+        }
+    },
+    # STAR WARS™ Jedi Knight II - Jedi Outcast™
+    "6030": {
+        "env": {
+            "MESA_EXTENSION_MAX_YEAR": "2003",
+            "__GL_ExtensionStringVersion": "17700",
+        }
+    },
+}
+
+
+class Tweaks:  # pylint: disable=too-few-public-methods
```

Comment:
Unless I'm missing something, this is unnecessary, you can just export `TWEAKS_DB` and do `env.update(TWEAKS_DB.get(env.get("SteamAppId")).get("env", {}))`

The pylint error about too few public methods is a hint :wink:

Reply:
Oh yes, I know - as I explained in PR, I am developing additional functionalities to this class, and this warning is already disabled in my local checkout :)

Reply:
This looks reasonably hard to read in such a packed form though

Reply:
@nanonyme I agree. Putting this line straight into `proton` file would also make it harder to implement unit tests in future. And Proton definitely could use some tests right now...

Reply:
```python
game_tweaks = Tweaks(env.get("SteamAppId"))
env.update(game_tweaks.env)
```
to

```python
game_tweaks = TWEAKS_DB.get(env.get("SteamAppId"), {})
env.update(game_tweaks.get("env", {}))
```

is as readable, and there are probably many other more readable possibilities.

Reply:
... unless I want to use `game_tweaks.import_fonts()` in some other place in code ;) But I hear you - maybe it would be better to implement future tweaks as more functions in `proton` file, but I wanted to separate responsibilities early. `proton` script is long and non-pythonic in several ways already.